src/main/java/seedu/address/model/task/Task.java

```diff
@@ -14,23 +15,39 @@ public class Task {
     private final Description description;
     private final boolean isTankRelatedTask;
     private Tank tank;
+    private Priority priority;
+    private boolean hasPriority;
 
     /**
      * Constructor for a Task
      * @param description Description of task
      * @param tank Tank (if applicable) tied to this task
+     * @param priority Priority (if applicable) tied to this task
      */
-    public Task(Description description, Tank tank) {
+    public Task(Description description, Tank tank, Priority priority) {
         this.description = description;
         this.tank = tank;
         this.isTankRelatedTask = this.tank != null;
+        this.priority = priority;
+        this.hasPriority = this.priority != null;
         this.isReminder = false;
     }
 
     public Description getDescription() {
         return this.description;
     }
 
+    public Priority getPriority() {
+        if (!hasPriority) {
+            throw new TaskHasNoPriorityException(this);
+        }
+        return this.priority;
+    }
+
+    public boolean hasPriority() {
+        return this.hasPriority;
+    }
+    
```

Comment:
I checked this part, there seems to be ' ' whitespace on the line after the hasPriority() method i.e if you click the right arrow key it doesn't immediately go to the next line but continues on the same line on the ' ' chars. Delete these ' ' chars and it should be fine